packages/tools/devtools/devtools-view/package.json

```diff
@@ -65,7 +65,7 @@
 		"test:mocha:verbose": "cross-env FLUID_TEST_VERBOSE=1 npm run test:mocha",
 		"test:screenshots": "npm run test:mocha",
 		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json && copyfiles -f ../../../../common/build/build-common/src/cjs/package.json ./dist",
-		"tsc:watch": "tsc --watch"
+		"tsc:watch": "copyfiles -f ../../../../common/build/build-common/src/cjs/package.json ./dist && fluid-tsc commonjs --watch --project ./tsconfig.cjs.json"
```

Comment:
No strong opposition to this, but I almost wonder if it would be better to mark `tsc` as a prerequisite to `tsc:watch` (i.e. add a `pretsc:watch` that runs `tsc`). That way we don't have to duplicate the file-copying logic.

Reply:
I have no general opposition to that pattern. I wonder about failure handling. The `tsc` scripts won't copy if `fluid-tsc` fails which is a real possibility when starting a watch. So, the pre would have to be just the copy or ???
There was a proposal to add the package.json placement smarts into fluid-tsc directly. I think that might be the best resolution. (We could enhance the package.json stub logic to handle the self-reference feature that is currently unsupported with our simple stub, too. I think only one package has needed that so far and has its own package.json that gets copied.) Feels lower priority until we need it. The copyfiles pattern is at least robust and works well with the incremental build tooling, despite not looking great.

Reply:
> The tsc scripts won't copy if fluid-tsc fails which is a real possibility when starting a watch.

Good point. Maybe just adding the copy step then. We could capture it in its own named script for reuse, and add `pretsc:watch` and `posttsc` that call it?

Reply:
Hmm. I am a little hesitant to use pre and post pattern, right now. We don't have https://pnpm.io/cli/run#enable-pre-post-scripts enabled. In pnpm 9.x the default is true, but it is defaulting to false in 8.x and earlier.
I don't know what it means for repo if we just switch to enable that.
"Run script | Debug script" from VS Code uses pnpm and currently skips pre+post scripts.
What we have now with [evil] duplication of a command across two script entries becomes 5 entries like:
```
		"place:cjs:package-stub": "copyfiles -f ../../../common/build/build-common/src/cjs/package.json ./dist",
		"posttsc": "npm run place:cjs:package-stub",
		"pretsc:watch": "npm run place:cjs:package-stub",
		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json",
		"tsc:watch": "fluid-tsc commonjs --project ./tsconfig.cjs.json --watch",
```
I am not a fan of duplication either but doesn't really feel better. One thing I don't like about pre+post is that they aren't sorted next to the thing that they impact which makes it harder to read what is going on. I think better use of pre+post is when the command is less under control (like pre/post install stuff) or not easy to be in one line.
Here if we really don't want to duplicate the copy line, then I think we can do:
```
		"place:cjs:package-stub": "copyfiles -f ../../../common/build/build-common/src/cjs/package.json ./dist",
		"tsc": "fluid-tsc commonjs --project ./tsconfig.cjs.json && npm run place:cjs:package-stub",
		"tsc:watch": "npm run place:cjs:package-stub && fluid-tsc commonjs --project ./tsconfig.cjs.json --watch",
```

Also, no rush to judgement as it turns out `fluid-tsc --watch` is broken. I thought I tested it way back in the day, but I guess I got confused, because support is definitely not there. I think I can fix it.

Reply:
Yeah, I'm good with this PR as is for now, just thinking about longer term improvements. And the potential short term improvement you suggested looks good to me too.

> One thing I don't like about pre+post is that they aren't sorted next to the thing that they impact which makes it harder to read what is going on.

Interesting. I forget which tool we're using for package.json sorting, but that seems like either an improvement we could upstream / something to investigate. If I remember correctly, `sort-package-json` _did_ sort `pre` and `post` scripts next to their corresponding core scripts.

Reply:
> Hmm. I am a little hesitant to use pre and post pattern, right now.

FWIW, we support lifecycle scripts in fluid-build. So they _should_ just work. But you can always define the fluid-build tasks explicitly.

> If I remember correctly, `sort-package-json` (what we used in `typescript-pipe`) _did_ sort `pre` and `post` scripts next to their corresponding core scripts.

That's what we use, and it does sort pre and post scripts alongside their "main" script but it only does so for well-known script names. https://github.com/keithamus/sort-package-json/blob/main/defaultRules.md#scripts